### ISSUE TITLE
TSX support

### ISSUE THREAD

**Issue by johannes-z** [egoist/bili#276]
I came across a problem, when compiling with bili and using render functions inside a tsx file.

1. The first proble mis, that the emitted code for `h` is `React.createElement`. I have this in my `tsconfig.json`: 

```json
    "jsx": "preserve",
    "jsxFactory": "h",
```

I found, that using this is needed for `bili.config.ts`:

```js
  babel: {
    configFile: false,
    jsx: 'h',
  },
```

2. Now that `React.createElement` is "replaced" by `h`, it still doesn't work if the initial `render` function calls other functions, that also return TSX, e.g. like this:

```jsx
render (h) {
  return (
    <div>Parent: {renderSubContent()}</div>
  )
},
renderSubContent() {
  return (
    <span>Content<span>
  )
}
```

In this case, `renderSubContent` throws an error, that `h is not defined`. If I add `const h = this.$createElement` manually to the function, there are no errors but also nothing gets rendered at all.

**Comment by egoist** [comment]
Yeah currently it only supports React JSX. We need to add an option to disable the built-in JSX plugin so that you can use Vue's babel plugin instead.
